gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java

```diff
@@ -0,0 +1,224 @@
+import com.google.cloud.storage.Storage.BlobGetOption;
+import com.google.cloud.storage.Storage.BlobSourceOption;
+import com.google.cloud.storage.Storage.BlobTargetOption;
+import com.google.cloud.storage.spi.RpcBatch;
+import com.google.cloud.storage.spi.StorageRpc;
+import com.google.common.collect.ImmutableMap;
+
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Map;
+
+public class StorageBatchTest {
+
+  private static final BlobId BLOB_ID = BlobId.of("b1", "n1");
+  private static final BlobId BLOB_ID_COMPLETE = BlobId.of("b1", "n1", 42L);
+  private static final BlobInfo BLOB_INFO = BlobInfo.builder(BLOB_ID).build();
+  private static final BlobInfo BLOB_INFO_COMPLETE = BlobInfo.builder(BLOB_ID_COMPLETE)
+      .metageneration(42L)
+      .build();
+  private static final BlobGetOption[] BLOB_GET_OPTIONS = {
+      BlobGetOption.generationMatch(42L), BlobGetOption.metagenerationMatch(42L)};
+  private static final BlobSourceOption[] BLOB_SOURCE_OPTIONS = {
+      BlobSourceOption.generationMatch(42L), BlobSourceOption.metagenerationMatch(42L)};
+  private static final BlobTargetOption[] BLOB_TARGET_OPTIONS = {
+      BlobTargetOption.generationMatch(), BlobTargetOption.metagenerationMatch()};
+  private static final GoogleJsonError GOOGLE_JSON_ERROR = new GoogleJsonError();
+
+  private StorageOptions optionsMock;
+  private StorageRpc dnsRpcMock;
+  private RpcBatch batchMock;
+  private StorageBatch dnsBatch;
+  private final Storage storage = EasyMock.createStrictMock(Storage.class);
+
+  @Before
+  public void setUp() {
+    optionsMock = EasyMock.createMock(StorageOptions.class);
+    dnsRpcMock = EasyMock.createMock(StorageRpc.class);
+    batchMock = EasyMock.createMock(RpcBatch.class);
+    EasyMock.expect(optionsMock.rpc()).andReturn(dnsRpcMock);
+    EasyMock.expect(dnsRpcMock.createBatch()).andReturn(batchMock);
+    EasyMock.replay(optionsMock, dnsRpcMock, batchMock, storage);
+    dnsBatch = new StorageBatch(optionsMock);
+  }
+
+  @After
+  public void tearDown() {
+    EasyMock.verify(batchMock, dnsRpcMock, optionsMock, storage);
+  }
+
+  @Test
+  public void testConstructor() {
+    assertSame(batchMock, dnsBatch.batch());
+    assertSame(optionsMock, dnsBatch.options());
+    assertSame(dnsRpcMock, dnsBatch.storageRpc());
+  }
+
+  @Test
+  public void testDelete() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<Void>> callback = Capture.newInstance();
+    batchMock.addDelete(EasyMock.eq(BLOB_INFO.toPb()), EasyMock.capture(callback),
+        EasyMock.eq(ImmutableMap.<StorageRpc.Option, Object>of()));
+    EasyMock.replay(batchMock);
+    StorageBatchResult<Boolean> batchResult = dnsBatch.delete(BLOB_ID.bucket(), BLOB_ID.name());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305501/issues/source?bid=3123103&fileBranchId=3247265#l98)